src/gentropy/dataset/study_locus_overlap.py

```diff
@@ -48,6 +51,36 @@ def from_associations(
         """
         return study_locus.find_overlaps()
 
+
+    def calculate_beta_ratio(self: StudyLocusOverlap) -> DataFrame:
+        """Calculate the beta ratio for the overlapping signals.
+
+        Returns:
+            DataFrame: A dataframe containing left and right loci IDs, chromosome
+            and the average sign of the beta ratio
+        """
+        expanded_overlaps = self.df.select("*", "statistics.*").drop("statistics")
```

Comment:
I don't have objection about the logic in this function, however it is written in a quite different way than the other bits of the code. I would suggest collapsing the expressions as follows:

```python
return (
    # Unpack statistics column:
    self.df.select("*", "statistics.*")
    .drop("statistics")
    # Drop any rows where the beta is null or zero
    .filter(
        f.col("left_beta").isNotNull() & 
        f.col("right_beta").isNotNull() &
        (f.col("left_beta") != 0) &
        (f.col("right_beta") != 0) 
    )
    # Calculate the beta ratio and get the sign, then calculate the average sign across all variants in the locus
    .withColumn(
        "betaRatioSign",
        f.signum(f.col("left_beta") / f.col("right_beta"))
    )
    # Aggregate beta signs:
    .groupBy("leftStudyLocusId","rightStudyLocusId","chromosome")
    .agg(
        f.avg("betaRatioSign").alias("betaRatioSignAverage")
    )
)
```